### PR TITLE
1.8/1443 fix credentials in stacktraces

### PR DESCRIPTION
**Pull request by charlesstrange2**
Slightly sad that i cant test exception logs to the file - `Log::shoudlReceive()` can't detect the way the exception handling system invokes it's own logger, rather than the Log interface. This means the mocker can't easily intercept. 

Have tried to track the decision to log full stacks or not with some reflection instead.

https://trello.com/c/zxWFWUfd/1443-fix-for-security-issue-from-14-11-2019-vvv-ee

### CODE REVIEW

File: app/Exceptions/Handler.php [hunk 1]
```diff
@@ -18,33 +28,79 @@ class Handler extends ExceptionHandler
...
+        OAuthServerException::class,
+    ];
+
+    /**
+     * Check an exception is worth a stacktrace
+     *
+     * @param Exception $e Exception
+     * @return bool;
+     */
+    protected function shallWeStackTrace(Exception $e)
```
Comment (katjam) [inline app/Exceptions/Handler.php]: Cute

File: tests/Unit/Exceptions/ExceptionHandlerTest.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+use Tests\StoreTestCase;
+
+class ExceptionHandlerTest extends StoreTestCase
+{
+
+    private $exceptions = [];
+
+    protected function setUp()
+    {
+        // Tests reach into the container, and get upset when run in bulk with other tests.
```
Comment (katjam) [inline tests/Unit/Exceptions/ExceptionHandlerTest.php]: :cry:

File: tests/Unit/Exceptions/ExceptionHandlerTest.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    /** @test */
+    public function itShouldStackTraceNotInProduction()
+    {
+        $handler = Container::getInstance()->make(Handler::class);
+        try {
+            foreach ($this->exceptions as $exception) {
+                // hashed to be "anything other than 'production'
+                Config::set('app.env', md5('production'));
```
Comment (katjam) [inline tests/Unit/Exceptions/ExceptionHandlerTest.php]: OK - but probs doesn't need to be so extreme. :)

Comment (charlesstrange2) [reply tests/Unit/Exceptions/ExceptionHandlerTest.php]: You're right, could have been just `local`

**Pull request merged**